### ISSUE TITLE
Compiling PHP with no-opt leads to invalid PHP7 code

### ISSUE THREAD

**Issue by digilist** [HaxeFoundation/haxe#9493]
By compiling some Haxe code to PHP with the `no-opt` compiler flag leads to the following function In the `Array_hx` class:

```php
	public function remove ($x) {
		#/usr/share/haxe/std/php/_std/Array.hx:127: lines 127-138
		$_gthis = $this;
		#/usr/share/haxe/std/php/_std/Array.hx:128: characters 3-22
		$result = false;
		#/usr/share/haxe/std/php/_std/Array.hx:129: lines 129-136
		$collection = $this->arr;
		foreach ($collection as $key => $value) {
			(function ($index, $value1)  use (&$x, &$_gthis, &$result) {
				#/usr/share/haxe/std/php/_std/Array.hx:130: lines 130-135
				if (Boot::equal($value1, $x)) {
					#/usr/share/haxe/std/php/_std/Array.hx:131: characters 5-39
					array_splice($_gthis->arr, $index, 1);
					#/usr/share/haxe/std/php/_std/Array.hx:132: characters 5-13
					$_gthis->length--;
					#/usr/share/haxe/std/php/_std/Array.hx:133: characters 5-18
					$result = true;
					#/usr/share/haxe/std/php/_std/Array.hx:134: characters 5-25
					break;
				}
			})($key, $value);
		}

		#/usr/share/haxe/std/php/_std/Array.hx:137: characters 3-16
		return $result;
	}
```

The break inside that function does not work and leads to a Fatal error since PHP 7:
```
PHP Fatal error:  'break' not in the 'loop' or 'switch' context in out/lib/Array_hx.php on line 449
```

To reproduce this, the actual Haxe code is irrelevant, it happens even with an empty Main class:
```hx
class Main {
}
```
```
haxe --php out --no-opt
```

When the `--no-opt` flag is removed the closure is removed and the code works fine.

**Comment by nadako** [comment]
I don't even know what `--no-opt` is supposed to do nowadays, maybe it should even be removed?

**Comment by digilist** [comment]
Not sure about that, I am playing around with Haxe for the first time. I am not really sure why I enabled it - probably because something was not working as expected and I wanted to try it out. So removing it is fine for me ;)

**Comment by lublak** [comment]
@nadako i think it was to disable the code optimizations
https://github.com/HaxeFoundation/haxe/blob/development/src/optimization/optimizer.ml#L331

foptimize is set by --no-opt to false

**Comment by RealyUniqueName** [comment]
This is already fixed in Haxe 4.1.0. See #9143
